test/types/models.test.ts

```diff
@@ -329,6 +331,37 @@ function gh11911() {
   });
 }
 
+function schemaInstanceMethodsAndQueryHelpers() {
+  type UserModelQuery = Query<any, HydratedDocument<User>, UserQueryHelpers> & UserQueryHelpers;
+  interface UserQueryHelpers {
+    byName(this: UserModelQuery, name: string): this
+  }
+  interface User {
+    name: string;
+  }
+  interface UserInstanceMethods {
+    findByName(name: string): Promise<HydratedDocument<User>>;
+  }
+  type UserModel = Model<User, UserQueryHelpers> & UserInstanceMethods;
```

Comment:
Seems a little odd that `UserInstanceMethods` is part of the `UserModel` type. You shouldn't be able to do `UserModel.findByName()`...

Reply:
You are correct, I think the confusion came from `TInstanceMethod` and `TStaticMethods` on Schema and `TMethodsAndOverrides` on Model which is the same as `TInstanceMethod`. I've renamed the interfaces in the test and added statics too.